src/prefect/infrastructure/kubernetes.py

```diff
@@ -67,11 +67,11 @@ class KubernetesJob(Infrastructure):
     # shortcuts for the most common user-serviceable settings
     image: str = Field(default_factory=get_prefect_image_name)
     namespace: str = "default"
-    service_account_name: str = None
+    service_account_name: Optional[str] = None
     image_pull_policy: Optional[KubernetesImagePullPolicy] = None
 
     # connection to a cluster
-    cluster_config: KubernetesClusterConfig = None
+    cluster_config: Optional[KubernetesClusterConfig] = None
```

Comment:
I also noticed that these types were incorrect, they should be optional since we're assigning `None` by default.

Reply:
Is it really optional though?

Reply:
It is since we're setting it to None, typeing wise at least.

Reply:
Ah yeah it is